autorel/changelog_generator/changelog_gen.py

```diff
@@ -0,0 +1,279 @@
+                    # When the commit is not part of this release
+                    pass
+
+    def _parse_commits_for_issue_nodes(self):
+        """
+            Parses the commits messages for the issue ids
+        """
+        issue_commit_list = []
+        for commit in self._commits:
+            # the commit node should be unvisited
+            if self._commit_map[commit.hex][1] is False:
+                parsed_data = self._parser.parse("issues", commit.message)
+                if parsed_data:
+                    if parsed_data[1] == "issue_ids":
+                        issue_ids = parsed_data[0]
+                        for issue_id in issue_ids:
+                            issue_commit_list.append(tuple((
+                                                        commit,
+                                                        issue_id
+                                                    )))
+                    # mark the commit node visited
+                    self._commit_map[commit.hex][1] = True
+        # Create changelog entries for all the independent issues
+        for element in issue_commit_list:
+            issue_id = element[1]
+            commit = element[0]
+            issue_obj = getIssue(issue_id,
+                                 commit=commit
+                                 )
+            self._add_changelog_entry(issue_obj)
+
+    def _extract_linked_issues(self):
+        """
+            Extract issues linked to pull requests
+        """
+        issue_pull_list = []
+        for pull_detail in self._pull_ids:
+            pull_id = pull_detail
+            pull_obj = getPull(pull_id)
+            parsed_data = self._parser.parse("issues", pull_obj.body)
+            if parsed_data:
+                if parsed_data[1] == "issue_ids":
+                    issue_ids = parsed_data[0]
+                    for issue_id in issue_ids:
+                        issue_pull_list.append(tuple((
+                                                pull_id,
+                                                issue_id
+                                              )))
+            else:
+                self._add_changelog_entry(pull_obj)
+
+        # Create changelog entries for all the issues liked with pull requests
+        for element in issue_pull_list:
+            issue_id = element[1]
+            pull_id = element[0]
+            issue_obj = getIssue(issue_id,
+                                 pull_id=pull_id
+                                 )
+            self._add_changelog_entry(issue_obj)
+
+    def generate(self):
+        """
+            Generate the changelog according to last provided tag
+        """
+        self._clone_repository()
+        self._generate_commit_map()
+        self._configure_parser()
+        self._parse_commits_for_merge_nodes()
+        self._mark_merged_commits()
+        self._parse_commits_for_issue_nodes()
+        self._extract_linked_issues()
+
+    def render(self):
```

Comment:
We should make the output format easily configurable. What do you think about moving the output format specification to `settings.py`? @lbudai